internal/process/queue.go

```diff
@@ -96,8 +131,43 @@ func (q *Queue) worker(queue workqueue.RateLimitingInterface, process func(key s
 				}
 
 				queue.Forget(key)
+				log.Infof("item for %s has been processed, no retry, element forgotten", id)
+
 				return false
 			}()
 		}
 	}
 }
+
+func (q *Queue) logAndUpdateWorkerTimes(key string, name string, log logrus.FieldLogger) {
+	// log time
+	now := time.Now()
+	lastTime, ok := q.workerExecutionTimes[name]
+	if ok {
+		log.Infof("execution - worker %s last execution time %s, executed after %s", name, lastTime, now.Sub(lastTime))
+	}
+	q.workerExecutionTimes[name] = now
+
+	log.Infof("processing item %s, queue length is %d", key, q.queue.Len())
+}
+
+func (q *Queue) HealthCheck() {
+	healthCheckLog := q.log.WithField("healthCheck", q.name)
+	var buffer bytes.Buffer
+	buffer.WriteString(fmt.Sprintf("health - queue length %d", q.queue.Len()))
+
+	for name, lastTime := range q.workerExecutionTimes {
+		timeSinceLastExecution := time.Since(lastTime)
+
+		buffer.WriteString(fmt.Sprintf(", [worker %s, last execution time: %s, since last execution: %s]", name, lastTime, timeSinceLastExecution))
+	}
+
+	healthCheckLog.Info(buffer.String())
+
+	for name, lastTime := range q.workerExecutionTimes {
+		timeSinceLastExecution := time.Since(lastTime)
```

Comment:
The same loop again. Code duplication.

Reply:
It is by design. I do not want warn messages appearing before worker times are logged.

Reply:
One loop and two buffers would suffice.

Reply:
Given the number of workers it is not the issue in terms of performance.
Clean code zealots would object.